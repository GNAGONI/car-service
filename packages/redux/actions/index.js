export * from './auctions';
export * from './longAuctions';
export * from './dropDowns';
export * from './car';
export * from './location';
export * from './user';
export * from './contactUs';
export * from './toasts';
export * from './quote';
