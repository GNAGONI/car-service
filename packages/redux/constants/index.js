import UserTypes from '../action-types/user';

export const STATUS = {
  IDLE: 'idle',
  RUNNING: 'running',
  READY: 'ready',
  SUCCESS: 'success',
  ERROR: 'error',
};

export const ignoredActionTypes = [
  UserTypes.USER_LOGIN.REQUEST,
  UserTypes.USER_SIGNUP.REQUEST,
  UserTypes.REFRESH_TOKEN.REQUEST,
  UserTypes.USER_LOGOUT.REQUEST,
  UserTypes.FORGOT_PASSWORD.REQUEST,
];

export const DELAY_TIME = 1000;

export const DEFAULT_POSTCODE = 'WC2N 5DU';

export const SUCCESS_BID = 'SUCCESS_BID';

export const SOME_ERROR_OCCURRED = 'There were errors. Please fix them and try again';
export const SUCCESSFULLY_LOGGED_IN = 'You have been successfully logged in';
export const SUCCESSFULLY_SIGNED_UP = 'You have successfully created an account on Car.service';
