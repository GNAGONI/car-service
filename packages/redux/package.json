{
   "name": "@car-service/redux",
   "version": "0.0.1-alpha.0",
   "private": true,
   "scripts": {
      "lint": "eslint . --ext .js --ext .jsx",
      "lint:fix": "yarn lint -- --fix",
      "lint:quiet": "yarn lint -- --quiet",
      "prettier": "prettier './**/*.js' --write --single-quote",
      "prettier:write": "yarn prettier -- --write",
      "prettier:check": "yarn prettier -- --list-different"
   },
   "dependencies": {
      "@car-service/api-client": "^0.0.1-alpha.0",
      "immutability-helper": "^3.0.0",
      "isomorphic-unfetch": "^3.0.0",
      "keymirror": "^0.1.1",
      "moment": "^2.24.0",
      "redux": "^4.0.1",
      "redux-actions": "^2.6.4",
      "redux-devtools-extension": "^2.13.8",
      "redux-saga": "^1.0.0",
      "reselect": "^4.0.0"
   },
   "devDependencies": {
      "@babel/cli": "7.1.5",
      "@babel/core": "7.1.6",
      "@babel/plugin-proposal-class-properties": "7.1.0",
      "@babel/plugin-proposal-object-rest-spread": "7.0.0",
      "@babel/plugin-transform-runtime": "7.1.0",
      "@babel/preset-env": "7.1.6",
      "@babel/preset-react": "7.0.0",
      "@babel/register": "7.0.0",
      "@babel/runtime": "7.1.5",
      "eslint": "^6.0.0",
      "prettier": "^1.18.2",
      "redux-devtools-extension": "^2.13.7"
   }
}
