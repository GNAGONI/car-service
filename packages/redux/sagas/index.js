import auctions from './auctions';
import auction from './auction';
import contactUs from './contactUs';
import browseByMake from './browseByMake';
import longAuctions from './longAuctions';
import liveAuction from './liveAuction';
import awaitingCollection from './awaitingCollection';
import pendingPayments from './pendingPayments';
import quotesArchive from './quotesArchive';
import carModelsAuction from './carModels';
import user from './user';
import toasts from './toasts';
import quote from './quote';
import scrappedCars from './scrappedCars';

export default [
  ...auctions,
  ...contactUs,
  ...auction,
  ...longAuctions,
  ...liveAuction,
  ...awaitingCollection,
  ...pendingPayments,
  ...quotesArchive,
  ...browseByMake,
  ...carModelsAuction,
  ...user,
  ...toasts,
  ...quote,
  ...scrappedCars,
];
