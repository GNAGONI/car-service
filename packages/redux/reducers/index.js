import { combineReducers } from 'redux';

import auctions from './auctions';
import auction from './auction';
import user from './user';
import userAuth from './userAuth';
import car from './car';
import longAuction from './longAuction';
import dropDown from './dropDown';
import liveAuction from './liveAuction';
import awaitingCollection from './awaitingCollection';
import pendingPayments from './pendingPayments';
import quotesArchive from './quotesArchive';
import modal from './modal';
import browseByMake from './browseByMake';
import location from './location';
import carModels from './carModels';
import message from './message';
import toasts from './toasts';
import quote from './quote';
import scrappedCars from './scrappedCars';

export default combineReducers({
  ...auctions,
  ...auction,
  ...car,
  ...user,
  ...longAuction,
  ...dropDown,
  ...liveAuction,
  ...awaitingCollection,
  ...pendingPayments,
  ...quotesArchive,
  ...modal,
  ...browseByMake,
  ...location,
  ...carModels,
  ...message,
  ...toasts,
  ...quote,
  ...scrappedCars,
  ...userAuth,
});
