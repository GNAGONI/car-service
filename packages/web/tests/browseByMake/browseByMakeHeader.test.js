import React from 'react'
import renderer from 'react-test-renderer'
import BrowseByMakeHeader from 'components/browseByMake/browseByMakeHeader'
import { connectTestComponentToRedux } from 'test/utils/redux'

it('Renders correctly', () => {
  const tree = renderer.create(connectTestComponentToRedux(<BrowseByMakeHeader />)).toJSON()

  expect(tree).toMatchSnapshot()
})
