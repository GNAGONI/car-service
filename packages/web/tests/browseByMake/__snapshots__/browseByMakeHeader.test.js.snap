// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="browse-by-make__header"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <nav
          className="subnav bg-before-default"
        >
          <ul />
        </nav>
      </div>
    </div>
  </div>
  <div
    className="header__content"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="pt-3"
          >
            <nav
              aria-label="breadcrumbs"
              className="nav-breadcrumb"
            >
              <ol
                className="breadcrumb"
              />
            </nav>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="pt-3 text-center"
          >
            <header
              className="section-head"
            >
              <div
                className="text-box mb-3"
              >
                <p />
              </div>
              <div
                className="header__selector-section"
              >
                <span
                  className="selector-section__title"
                >
                  Browse car makes
                </span>
                <div
                  className="selector-section__wrapper"
                >
                  <div
                    className="css-btephf"
                    onKeyDown={[Function]}
                  >
                    <div
                      className="css-55nwcx"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className="css-cys4jg"
                      >
                        <div
                          className="css-1492t68"
                        >
                          
                        </div>
                        <input
                          className="css-14uuagi"
                          disabled={false}
                          id="react-select-2-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          readOnly={true}
                          tabIndex="0"
                          value=""
                        />
                      </div>
                      <div
                        className="css-1wy0on6"
                      >
                        <span
                          className="css-rb20kt"
                        />
                        <div
                          aria-hidden="true"
                          className="css-1ep9fjw"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-19bqh2r"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </header>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
